contracts/eosio.system/src/delegate_bandwidth.cpp

```diff
@@ -123,16 +123,25 @@ namespace eosiosystem {
 
    /**
     * This action will transfer RAM bytes from one account to another.
-   */
-   void system_contract::ramtransfer( const name& from, const name& to, int64_t bytes ) {
+    */
+   void system_contract::ramtransfer( const name& from, const name& to, int64_t bytes, const std::string& memo ) {
       require_auth( from );
       update_ram_supply();
+      check( memo.size() <= 256, "memo has more than 256 bytes" );
       reduce_ram( from, bytes );
       add_ram( to, bytes );
       require_recipient( from );
       require_recipient( to );
    }
 
+   /**
+    * This action will burn RAM bytes from owner account.
+    */
+   void system_contract::ramburn( const name& owner, int64_t bytes, const std::string& memo ) {
+      require_auth( owner );
+      ramtransfer( owner, null_account, bytes, memo );
```

Comment:
@ericpassmore this is the only logic for `ramburn`

Not sure if we need extensive testing for the `ramburn` specific action

Reply:
How about checking ram balence of `eosio.null` account before and after `ramburn` making sure the change in balance matches the amount passed in `ramburn`?